src/Illuminate/Database/Query/Grammars/SqlServerGrammar.php

```diff
@@ -200,6 +204,19 @@ public function compileTruncate(Builder $query)
     }
 
     /**
+     * Compile an exists statement into SQL.
+     *
+     * @param \Illuminate\Database\Query\Builder $query
+     * @return string
+     */
+    public function compileExists(Builder $query)
+    {
+        $select = $this->compileSelect($query);
+
+        return "select cast(case when exists($select) then 1 else 0 end as bit) as {$this->wrap('exists')}";
```

Comment:
No. It stops as soon as soon as logical test proves true. http://sqlmag.com/t-sql/exists-vs-count